### PR TITLE
Rid of lodash

### PR DESCRIPTION
**Pull request by ufocoder**
#11

### CODE REVIEW

File: services/fhir.ts [hunk 1]
```diff
@@ -225,35 +225,39 @@ export function makeReference<T extends AidboxResource>(
...
-            'resource',
-            'display',
-            'identifier',
-            'uri',
-            'localRef',
-            'extension',
-        ])
-    );
+    return !Object.keys(resource).filter(
+        (resource) =>
```
Comment (ir4y) [inline services/fhir.ts]: Use the `attribute` variable name instead of `resource`.

**Comment by ir4y** [comment]
@ruscoder @m0rl 
Please elaborate, add a set of lodash functions you use in daily tasks.
We need to backport it via native primitives.

**Comment by dmitryashutov** [comment]
I use often:

File: services/fhir.ts [hunk 2]
```diff
@@ -225,35 +225,39 @@ export function makeReference<T extends AidboxResource>(
...
 }
 
 export function getIncludedResource<T extends AidboxResource>(
     // TODO: improve type for includedResources: must contain T
     resources: ResourcesMap<T | any>,
     reference: AidboxReference<T>
 ) {
-    return _.find<T>(resources[reference.resourceType], (resource) => resource.id === reference.id);
+    const typeResources = resources[reference.resourceType];
+    const index = typeResources ? typeResources.findIndex((resource: T) => resource.id === reference.id) : -1;
```
Comment (ruscoder) [inline services/fhir.ts]: let's rewrite it
```
if (!typeResources) {
    return undefined;
}
const index = typeResources.findIndex((resource: T) => resource.id === reference.id);
return typeResources[index];
```

File: services/fhir.ts [hunk 3]
```diff
@@ -225,35 +225,39 @@ export function makeReference<T extends AidboxResource>(
...
 }
 
 export type ResourcesMap<T extends AidboxResource> = { [x: string]: T[] | undefined };
 
 export function extractBundleResources<T extends AidboxResource>(bundle: Bundle<T>): ResourcesMap<T> {
-    const entriesByResourceType = _.groupBy(bundle.entry, (entry) => entry.resource!.resourceType);
+    const entriesByResourceType = {};
 
-    return _.mapValues(entriesByResourceType, (entries) => _.map(entries, (entry) => entry.resource!));
+    bundle.entry?.forEach(function(entry) {
```
Comment (ruscoder) [inline services/fhir.ts]: bundle.entry? -- what does it mean?

Comment (ruscoder) [reply services/fhir.ts]: By the way, bundle.entry can be undefined following the spec. Let's check it before.

File: services/fhir.ts (hunk 1 above)
Comment (ruscoder) [inline services/fhir.ts]: wrong naming. instead of resource in filter use key

File: hooks/pager.ts [hunk 4]
```diff
@@ -35,7 +35,9 @@ export function usePager<T extends AidboxResource>(
         resources,
         {
             loadNext: () => setPageToLoad((currentPage) => currentPage + 1),
-            hasNext: isSuccess(resources) && !!_.find(resources.data.link, { relation: 'next' }),
+            hasNext:
+                isSuccess(resources) &&
+                Boolean(resources.data.link && resources.data.link.some((link) => link.relation == 'next')),
```
Comment (ruscoder) [inline hooks/pager.ts]: We usually use === and !== by default for all conditions (it makes life easier because you don't need to think about how it works)

**Comment by ruscoder** [comment]
> @ruscoder @m0rl
> Please elaborate, add a set of lodash functions you use in daily tasks.
> We need to backport it via native primitives.

I'm not sure that we need to do it in this way:
1. A lot of lodash functions can be useful
2. Native primitives worked not so fast in comparison to lodash alternatives (a year ago, now - I don't know)
3. We can override type definitions for some lodash functions we use
4. Let's do it in another way, as we discussed with @dymio yesterday.


Proposed solution (project related! not aidbox-react package):
1. Create file src/utils/lodash.ts (for example)
2. Import the particular lodash functions using the function package, e.g. `import map from 'lodash/map';`
3. Export them

And use in the project in the following way:
```
import * as _ from 'src/utils/lodash';

_.map()
```

It will reduce the bundle size, and we can easily add new lodash function to the project.

@ir4y What do you think?

File: __tests__/services/fhir.spec.ts [hunk 5]
```diff
@@ -308,6 +308,8 @@ describe('Service `fhir`', () => {
                 extraField: true,
             })
         ).toBeFalsy();
+
+        expect(isReference({} as any)).toBeTruthy();
```
Comment (ruscoder) [inline __tests__/services/fhir.spec.ts]: Should be false

**Pull request merged**